quinn-proto/src/connection/mod.rs

```diff
@@ -477,78 +585,87 @@ where
                 prev.update_unacked = false;
             }
 
-            let mut builder = self.begin_packet(
+            debug_assert!(
+                builder.is_none() && sent_frames.is_none(),
+                "Previous packet must have been finished"
+            );
+
+            builder = Some(self.begin_packet(
                 now,
                 space_id,
-                pad_space == Some(space_id),
                 &mut buf,
                 buf_capacity,
-            )?;
+                (num_datagrams - 1) * (self.path.mtu as usize),
+                ack_eliciting,
+            )?);
+            let builder = builder.as_mut().unwrap();
             coalesce = coalesce && !builder.short_header;
 
-            let sent = if close {
+            sent_frames = if close_in_space {
                 trace!("sending CONNECTION_CLOSE");
                 match self.state {
                     State::Closed(state::Closed { ref reason }) => {
                         if space_id == SpaceId::Data {
-                            reason.encode(&mut builder.buffer, builder.max_size)
+                            reason.encode(&mut &mut buf, builder.max_size)
```

Comment:
Do we actually need `&mut &mut` here?

Reply:
We don't. Was a rename artifact - thanks for catching!